actors/builtin/reward/reward_actor.go

```diff
@@ -43,9 +38,9 @@ func (r *Reward) AmountVested(elapsedEpoch abi.ChainEpoch) abi.TokenAmount {
 	case None:
 		return r.Value
 	case Linear:
-		TODO() // zx BigInt
-		vestedProportion := math.Max(1.0, float64(elapsedEpoch)/float64(r.StartEpoch-r.EndEpoch))
-		return big.Mul(r.Value, big.NewInt(int64(vestedProportion)))
+		vestDuration := big.Sub(big.NewInt(int64(r.EndEpoch)), big.NewInt(int64(r.StartEpoch)))
+		unitVest := big.Div(r.Value, vestDuration)
```

Comment:
It's a nit, but when value (in attofil) is less than duration (in epochs), the vesting rate of < 1aF/epoch rounds to zero. This will happen when block rewards approach zero, (at which point we don't really need vesting).

The most correct solution is to rearrange arithmetic so the division happens _last_ (after the mul below).
Alternatively, add a guard before this to vest fully when the rate is <1 (generous) or vest fully when elapsed > duration (conservative).